R/recipes.R

```diff
@@ -31,7 +31,10 @@ model_main_recipe <- function(data, pred_vars, cat_vars,
     # Remove any variables not an outcome var or in the pred_vars vector
     step_rm(-all_outcomes(), -all_predictors(), -has_role("ID")) %>%
     # Impute missing values using KNN. Specific to condo model, usually used to
-    # impute missing condo building strata
+    # impute missing condo building strata. Within step_impute_knn, an estimated
+    # node value is called with the sample(). This is not deterministic, meaning
+    # different runs of the model will have different imputed values, and thus
+    # different FMVs.
```

Comment:
**issue (blocking):** This is _sort of_ correct. The imputation will be deterministic between model runs _if_ a seed is set somewhere in this file. However, it _won't be_ deterministic if you run the same prediction twice in a single session (unless you set the seed again).

I would add `set.seed(params$input$strata$seed)` somewhere at the top of this file to ensure that prediction is always using the same seed. Then run the stage twice (run once, restart, run again) and check that the results are the same.

Reply:
I included this in the file as described, and as setting a global seed. In each of those situations, different iterations still created different FMVs. You can use the uploaded testing file to look into this.